titan/features/prep.py

```diff
@@ -0,0 +1,333 @@
+from typing import Dict, ClassVar, Set
+
+from . import base_feature
+from .. import agent
+from .. import population
+from .. import model
+from ..parse_params import ObjMap
+
+
+class Prep(base_feature.BaseFeature):
+
+    name = "prep"
+    stats = ["prep", "prep_new", "prep_injectable", "prep_oral"]
+    """
+        PrEP collects the following stats:
+
+        * prep - number of agents with active PrEP
+        * prep_new - number of agents who became active PrEP this time step
+        * prep_injectable - number of agents on injectable PrEP
+        * prep_oral - number of agents on oral PrEP
+    """
+
+    # class level attributes to track all Prep agents
+    counts: ClassVar[Dict[str, int]] = {}
+    new_agents: ClassVar[Set["agent.Agent"]] = set()
+
+    def __init__(self, agent: "agent.Agent"):
+        super().__init__(agent)
+        # agent level attributes
+        self.active = False
+        self.adherence = 0
+        self.type = ""
+        self.load = 0.0
```

Comment:
We might be able to get rid of prep load and condense that with the probability equation in `hiv_convert`. It's essentially just a middle step between time since injection and the risk calculation.

Reply:
I'd imagine saving `last_dose` instead would also be useful for potential outputs during projects? If noted for both LAI and oral prep.

Reply:
perhaps we should move those risk adjustment functions into the features themselves - maybe add methods like `adjust_partner_risk(self)` and `adjust_agent_risk(self)` or maybe `get_risk_adjustment(self, partner=True/False)` with a default of returning `1.0`  - this is kind of hiv centric is my only hesitancy, so we'd have to figure out that interaction if/when we abstract disease.  thoughts?

Reply:
or maybe `get_acquisition_risk_multiplier` and `get_transmission_risk_multiplier`

Reply:
Would the risk adjustment necessarily be HIV-centric? I like the `get` idea.